### DISCUSSION TITLE
Discussion on future ideas and other stuff to implement

### DISCUSSION THREAD

**Discussion by gaecoli** [getredash/redash#6360] [General]
We do so much about code format, code dependencies, I think we can stop these efforts for a while and start doing something that is really useful for functionality and performance, instead of staying on the format, which is a wrong direction. @justinclift

**Comment by justinclift** [comment]
@gaecoli You are totally welcome to start thinking about the future direction(s) for the project, discussing it, etc.

Should we do something like make a list of "possible" ideas, then we vote on which ones to do first, which ones should be "maybe later", and so on?

Also - I'll personally probably still focus on fixing the security problems, until they're all solved.  But that shouldn't stop you and everyone else wanting to do other stuff (like this). :smile:

**Reply by grugnog** [reply]
Thanks for working on the security updates too :)

**Reply by justinclift** [reply]
No worries.  That's the main thing I personally want solved in the short term too. :smile:

**Comment by gaecoli** [comment]
Yes, you are right!

**Reply by eradman** [reply]
The effort to update dependancies was very important. The changes we have made so far seems solid; my vote is to cut the first "community release" and then move on to changes that will require more time to test.

**Comment by justinclift** [comment]
Just re-opened this discussion, as it seems like a decent place to discuss idea for future things. :smile:

**Comment by justinclift** [comment]
Ideas wise, this is the stuff that interests me personally.  So these are things I'll probably be putting my personal time into:

#### 1. Improving operational aspects of Redash for people that run it

First step for that is to have the "production" setup script (in the "setup" repo) also set up automatic backups.  Nightly at the minimum.

Maybe leaving the backups in some filesystem location for people to automatically transfer off the host each night (along with the matching /opt/redash/env file).

That will need boldly documenting in some kind of "Production deployment steps" guide.

This is because having someone accidentally kill their Redash instance shouldn't mean for a complete loss, like happened here: https://github.com/getredash/redash/discussions/6135

#### 2. Look into moving the Redash data sources / query runners out of the Redash repo, into their own repos

This is to do two main things:

* Reduce the number of Python dependencies in the main repository (by a lot)
* Create some kind of (internal?) API that query runners use for receiving instructions from Redash and responding with results
  * This should also mean query runners can be written in any language.
  * ... and their dependencies are their own problem, not something the main Redash repo needs to care about.

This one is only a very rough concept in my head so far.  Needs a lot more thinking done for it. :smile:

---

These are my personal list only.  They shouldn't stop or block anyone else from having their own ideas on what they want to get done. :smile:

**Comment by konnectr** [comment]
To prioritize ideas , you can use this list )
https://github.com/getredash/redash/issues?q=is%3Aissue+is%3Aopen+sort%3Acomments-desc
and https://github.com/getredash/redash/pulls?q=is%3Apr+is%3Aopen+sort%3Acomments-desc

**Reply by justinclift** [reply]
Heh Heh Heh

Looking at the "Issues" list there... that's a *lot* of pages that need looking through.

It's probably not wise for us to go through those issues *before* trying to discuss future ideas.

I mean, they are probably important things in there... however there's 500+ issues. (!!!)  Let's think about reducing them later on, after we have an initial Community release created and known to be stable. :smile:

**Reply by rpmcoding** [reply]
@justinclift  Yes 500+ issues is just not manageable.

Do we have any insights on how many redash deployments are currently in production?  And any visibility into how many folks used to be on the paid cloud version before it was shuttered?  Wondering if we can reignite that as a way to drive a little revenue to pay for tools and time etc.

Also, perhaps something like canny.io but opensource would be useful for us to start collecting the bugs and feature requests that matter the most directly from users in order to prioritise.  They can up and down vote them rather than just 500+ issues which are not ordered.

**Reply by justinclift** [reply]
Some kind of voting thing would be useful.  Bearing in mind that even with that we'd have to figure out which of the 500+ issues is something that would need to go into the "pool of things to vote on". :wink:

Haven't come across canny.io before, do you reckon it might be useful for us?  As you say, it doesn't seem to be OSS (which would be nice).  But they *might* have some kind of "free for OSS use" type of plan available (no idea), which at least might make it an option. :smile:

**Reply by arikfr** [reply]
canny.io seem to have a free plan, but I don't know if it's usable enough for our needs. Their next plan is $400/mo 😮 I think I saw similar products with more affordable pricing (just in case we grow of the free plan). Also GitHub has a voting of sorts (I think you can sort the issues by comments & reactions).

> Do we have any insights on how many redash deployments are currently in production? And

We collect information from version checks which gives some visibility into how many deployments are there + what versions. This is optional so not 100% coverage but gives an idea. I need to remove the dust from it and make sure it's still live. 

> And any visibility into how many folks used to be on the paid cloud version before it was shuttered?

Several hundred companies (vs. thousands of OSS deployments). 
But I'm not sure I would recommend setting up SaaS again as a way to drive revenue.

**Reply by guidopetri** [reply]
> But I'm not sure I would recommend setting up SaaS again as a way to drive revenue.

Why not?

**Reply by justinclift** [reply]
From my point of view, Redash in its current form is incredibly resource (memory) wasteful.  Its memory usage needs to be reduced by at least an order of magnitude in order for hosting of it to be practical.

Saying that from already hosting a few paying customers.  Redash has an unholy memory footprint (:wink:) when customers start pushing it.  Thus my strong desire to solve this problem (eg potential use of Go based query runners and similar), once we get the current source code back to a good state.

**Comment by gaecoli** [comment]
> From my point of view, Redash in its current form is incredibly resource (memory) wasteful. Its memory usage needs to be reduced by at least an order of magnitude in order for hosting of it to be practical.
> 
> Saying that from already hosting a few paying customers. Redash has an unholy memory footprint (😉) when customers start pushing it. Thus my strong desire to solve this problem (eg potential use of Go based query runners and similar), once we get the current source code back to a good state.

Very interested in this part.

**Comment by robosong** [comment]
I'm new to github and sadly not a developer but I am a regular user and promoter of Redash in my company. I just joined github to post a couple ideas/requests for Redash features and noticed this discussion. With regard to a process for proposing/voting on future ideas, isn't there already a mechanism for this in github: https://github.com/getredash/redash/discussions/categories/ideas

Another software project I use, Dynalist, uses Trello to engage their userbase to propose and vote on new features and determine their development roadmap and it seems to be quite an effective process for them https://trello.com/b/z0HxDPNo/dynalist-roadmap

Apologies if I'm missing something and this isn't useful. I love Redash and I'm excited to see further development and grateful to those working on it. Cheers.
